src/ros_can_nodes/src/CANMsgRouter.cpp

```diff
@@ -13,24 +13,30 @@
 #include <chrono>
 #include <string>
 #include <stdexcept>
+#include "IntrospectionHelpers.hpp"
 #include "CANMsgRouter.hpp"
 #include "ROSCANConstants.hpp"
 #include "TopicBuffers.hpp"
 #include "CANHelpers.hpp"
 #include "RosCanNode.hpp"
 #include "RosCanNodeManager.hpp"
-
+#include "message_properties_map.hpp"
 
 int main(int argc, char **argv) {
     CANMsgRouter::init();
 
     //CANMsgRouter::subscriberTest();
-    // CANMsgRouter::publisherTest();
+    //CANMsgRouter::publisherTest();
 
     CANMsgRouter::run();
 }
 
 void CANMsgRouter::init() {
+    // preload registering of all possible messages
+    for (const auto& m: message_properties_map) {
+        IntrospectionHelpers::register_message(m.first, m.second.definition);
```

Comment:
should we do this when the topics get registered - it looks like this could use a lot of resources.

Reply:
ah yea, i'll make it so that it registers the message when a control message for subscribe/advertise arrives